src/config.rs

```diff
@@ -262,15 +335,606 @@ pub fn write_configs_to_path(cfgs: &[&WireguardConfig], path: &Path) -> io::Resu
+
+    let combined_errors = if errors.is_empty() {
+        None
+    } else {
+        Some(errors.join("\n"))
+    };
+
+    (scripts, combined_errors)
+}
+
+/// Validate the script content.
+/// Returns `Ok(())` if valid, otherwise `Err(String)` describing the issue.
+/// Validate that the script contains at least one of the required keywords.
+/// Parse the script content and extract routing keywords
+fn parse_routing_keywords(content: &str, script_name: &str) -> Result<RoutingHooks> {
+    let mut pre_up: Option<String> = None;
+    let mut post_up: Option<String> = None;
+    let mut pre_down: Option<String> = None;
+    let mut post_down: Option<String> = None;
+    let mut fwmark: Option<String> = None;
+    let mut has_bind_interface = false;
+
+    for raw_line in content.lines() {
+        let line = raw_line.trim();
+        if line.is_empty() || line.starts_with('#') {
+            continue;
+        }
+
+        // Split line into key/value
+        let (key, cmds_raw) = line.split_once('=').ok_or_else(|| {
+            anyhow!(
+                "Invalid syntax in script '{}': missing '=' in line '{}'",
+                script_name,
+                line
+            )
+        })?;
+
+        // Convert string to enum
+        let keyword = key.trim().parse::<RoutingKeyword>().map_err(|_| {
+            anyhow!(
+                "Unknown routing keyword '{}' in script '{}'",
+                key.trim(),
+                script_name
+            )
+        })?;
+
+        let cmds_raw = cmds_raw.trim();
+        let parts: Vec<&str> = cmds_raw
+            .split(';')
+            .map(|c| c.trim())
+            .filter(|c| !c.is_empty())
+            .collect();
+
+        if parts.is_empty() {
+            anyhow::bail!(
+                "Empty command list for {:?} in script '{}'",
+                keyword,
+                script_name
+            );
+        }
+
+        if keyword != RoutingKeyword::FwMark {
+            for cmd in &parts {
+                if !(cmd.starts_with("iptables")
+                    || cmd.starts_with("ip ")
+                    || cmd.starts_with("ip6tables"))
+                {
+                    anyhow::bail!(
+                        "Invalid command '{}' for {:?} in script '{}'. Only iptables/ip/ip6tables allowed.",
+                        cmd,
+                        keyword,
+                        script_name
+                    );
+                }
+                if cmd.contains("%bindIface") {
+                    has_bind_interface = true;
+                }
+            }
+        }
+
+        let joined = parts.join("; ");
+
+        match keyword {
+            RoutingKeyword::PreUp => pre_up = Some(joined),
+            RoutingKeyword::PostUp => post_up = Some(joined),
+            RoutingKeyword::PreDown => pre_down = Some(joined),
+            RoutingKeyword::PostDown => post_down = Some(joined),
+            RoutingKeyword::FwMark => fwmark = Some(joined),
+        }
```

Comment:
Just wondering if it would make sense to warn if a value is redefined, something like:

```rust
    if match keyword {
        RoutingKeyword::PreUp => pre_up.replace(joined),
        ...
    }.is_some() {
        warn!("{key} redefined, ignoring previous value", key = key.trim());
    }
```